src/main/java/io/projectriff/samples/shopping/inventoryapi/article/ArticleRepository.java

```diff
@@ -10,7 +10,16 @@
 @RepositoryRestResource(path = "article")
 public interface ArticleRepository extends CrudRepository<Article, Long> {
 
+  @Query("SELECT id,sku,name,description,price_in_usd,image_url,quantity FROM article WHERE sku = :sku")
+  Article findBySku(@Param("sku") String sku);
+
+  @RestResource(exported = false)
+  @Modifying
+  @Query("UPDATE article SET quantity = :to WHERE sku = :sku AND quantity = :from")
```

Comment:
now that I think about it, is there a particular reason for requiring the current quantity in the query?

Reply:
We only want to change the quantity if it hasn't been updated by someone else while we were processing, so checking that it still is what we think it is seems proper to achieve optimistic concurrency control